server/src/services/formation.service.ts

```diff
@@ -0,0 +1,870 @@
+
+  const esQueryIndexFragment = getFormationEsQueryIndexFragment(limit, options)
+
+  const responseFormations = await esClient.search({
+    ...esQueryIndexFragment,
+    body: {
+      query: {
+        bool: {
+          must: mustTerm,
+        },
+      },
+    },
+  })
+
+  const formations = []
+
+  responseFormations.body.hits.hits.forEach((formation) => {
+    formations.push({ source: formation._source, sort: formation.sort, id: formation._id })
+  })
+
+  if (formations.length === 0 && !caller) {
+    await notifyToSlack({ subject: "FORMATION", message: `Aucune formation par région trouvée pour les romes ${romes} ou le domaine ${romeDomain}.` })
+  }
+
+  return formations
+}
+
+/**
+ * Tente de récupérer des formations dans le rayon de recherche, si sans succès cherche les maxOutLimitFormation les plus proches du centre de recherche
+ * @param {string[]} romes un tableau de codes ROME
+ * @param {string} romeDomain un domaine de ROME
+ * @param {[number, number]} coords des coordonnées au format [longitude,latitude] pour centrer la recherche
+ * @param {number} radius le rayon de recherche
+ * @param {string} diploma le niveau de diplôme visé
+ * @param {number} limit le nombre de résultats max à produire
+ * @param {string} caller l'identifiant fourni par l'exploitant de l'api
+ * @param {string} api le nom de l'api liée à cette fonction
+ * @param {string[]} options un tableau d'options modulant la recherche
+ * @param {string} useMock un flag indiquant s'il faut retourner des données mockées ou non
+ * @returns {Promise<ILbaItem[]>}
+ */
+const getAtLeastSomeFormations = async ({
+  romes,
+  romeDomain,
+  coords,
+  radius,
+  diploma,
+  maxOutLimitFormation,
+  caller,
+  options,
+  useMock,
+}: {
+  romes?: string[]
+  romeDomain?: string
+  coords?: [string, string]
+  radius?: number
+  diploma?: string
+  maxOutLimitFormation?: number
+  caller?: string
+  options: string[]
+  useMock: string
+}): Promise<ILbaItem[]> => {
+  let rawEsFormations: IFormationEsResult[] = null
+  let currentRadius = radius
+  let formationLimit = formationResultLimit
+
+  rawEsFormations = await getFormations({
+    romes,
+    romeDomain,
+    coords,
+    radius: currentRadius,
+    diploma,
+    limit: formationLimit,
+    caller,
+    options,
+    useMock,
+  })
+
+  // si pas de résultat on étend le rayon de recherche et on réduit le nombre de résultats autorisés
+  if (rawEsFormations instanceof Array && rawEsFormations.length === 0) {
+    formationLimit = maxOutLimitFormation // limite réduite car extension au delà du rayon de recherche
+    currentRadius = 20000
+    rawEsFormations = await getFormations({
+      romes,
+      romeDomain,
+      coords,
+      radius: currentRadius,
+      diploma,
+      limit: formationLimit,
+      caller,
+      options,
+      useMock,
+    })
+  }
+
+  rawEsFormations = deduplicateFormations(rawEsFormations)
+
+  const formations: ILbaItem[] = transformFormationsForIdea(rawEsFormations)
```

Comment:
```suggestion
  const formations = transformFormationsForIdea(rawEsFormations)
```
la fonction parent renvoie bien le type, laisser l'inférence typer la variable